Fixes #32438 - tmuxinator and common startup scripts

lzap opened the pull request:
This patch unifies starting The Foreman via both foreman gem and a shell script. We used to have two entrypoints: Procfile (foreman) and scripts/foreman-start-dev. This is now all replaced with two new very simple scripts start-rails and start-webpack. In Procfile we simply call these.

On top of that, it provides project configuration for tmuxinator gem, a tool to manage tmux sessions via YAML configuration. It is similar to my previous PR ("foreman-cmd") but using a specialized tool makes the patch and code we need to carry very low.

Therefore with this solution, people can choose if they want to keep using "foreman" gem or they want terminal multiplexer (which is REPL/debugger friendly). Also both Procfile and .tmuxinator.yaml are small and startup scripts are clean and the single space holding the startup logic. Optionally, configuration file can be created with custom things like different port etc.

See https://github.com/theforeman/foreman/pull/8476 and https://github.com/tmuxinator/tmuxinator

lzap committed:
Fixes #32438 - tmuxinator and common startup scripts

lzap commented:
We had a chat on the maintainers meeting, here are TODO points:

* move whatever can be moved into bin/ (except Procfile and yaml config)
* change port to be always 3000 (this will break foreman gem startup but people agreed we should have just a single port)
* update docs about the port change and alternative way of starting

ekohl reviewed (changes requested):
What `foreman` does is read `.env`. I think you shouldn't introduce a new convention with `.start.conf.sh`. https://github.com/theforeman/puppet-katello_devel/blob/master/templates/env.erb is the template that's used in the katello devel box. Also, perhaps we can take this chance to clean it up.

https://stackoverflow.com/questions/28668436/how-to-change-the-default-binding-ip-of-rails-4-2-development-server/29562898#29562898 suggests you can change the default bind in boot. If we prefer `::`, perhaps that's a good thing to do. We could also take a look at unifying binding with production:
https://github.com/theforeman/foreman/blob/5c3d913c14746bae84d90bdfd8c9fb11d3cf4017/config/puma/production.rb#L33-L47

lzap commented:
The `.env` is a good idea, I will rewrite this. Unfortunately `.env*` is in `.gitignore` so I cannot provide `.env.example` but I figured out a good solution - startup scripts will create the `.env` file automatically on the first run so users do not need to figure anything.

Renamed the variables to match what we use in the Foreman codebase too. Rebased.

lzap commented:
I have just added the new ENV variable for webpack introduced in 6251a8d59e274da73300c2fd94da33e0504552ee

ezr-ondrej commented:
So this would be a nice to have, but unless someone else then me & lzap (neighter of us will be maintaining it long term) sees a value in this, I propose closing this as what we have now kinda works, even tho debugging is still pain.
Looking at you @adamruzicka would you see a value in finishing this up ? :)

lzap commented:
I ended up using this less and less. Closing.
